website/source/docs/auth/aws-iam.html.md

```diff
@@ -0,0 +1,606 @@
+---
+layout: "docs"
+page_title: "Auth Backend: AWS-IAM"
+sidebar_current: "docs-auth-aws-iam"
+description: |-
+  The aws-iam backend allows automated authentication AWS IAM principals.
+---
+
+# Auth Backend: aws-iam
+
+The aws-iam auth backend provides a secure introduction mechanism for AWS IAM
+principals. This allows you to reduce the problem of securely introducing a
+Vault token to the problem of securely introducing AWS IAM credentials, which
+AWS has already solved in a number of use cases, such as via EC2 Instance
+Profiles and IAM roles attached to Lambda functions. It also allows you to have
+a consistent workflow between developers working on a local laptop with tools
+such as Hologram.
+
+## Comparison with AWS-EC2 Authentication Backend
+
+The AWS-IAM and AWS-EC2 authentication backends serve similar purposes. Both
```

Comment:
Thanks for the refresher, and sorry for not remembering the details, Honestly, I never saw any of those additional attributes as "interesting" from a security perspective. AWS has never really made anything other than IAM instance profile as a security sensitive attribute in launching an instance (e.g., with the very specific iam:PassRole permission). In theory, many of these could also be implemented (since the instance ID gets returned from the `GetCallerIdentity` call, assuming that only EC2 is able to AssumeRole the role in question) with the `aws-iam` backend, but they haven't yet (though could be in a future PR).

I'll work on a documentation update to be more clear about these points.